packages/framework-provider-aws/src/index.ts

```diff
@@ -57,63 +9,66 @@ function loadInfrastructurePackage(packageName: string): HasInfrastructure {
  * to extend the AWS functionality. Rockets are typically distributed in separate node packages.
  */
 export const Provider = (rockets?: RocketDescriptor[]): ProviderLibrary => {
-  return {
-    // ProviderEventsLibrary
-    events: {
-      rawToEnvelopes: rawEventsToEnvelopes,
-      forEntitySince: readEntityEventsSince.bind(null, dynamoDB),
-      latestEntitySnapshot: readEntityLatestSnapshot.bind(null, dynamoDB),
-      search: searchEvents.bind(null, dynamoDB),
-      store: storeEvents.bind(null, dynamoDB),
-    },
-    // ProviderReadModelsLibrary
-    readModels: {
-      rawToEnvelopes: rawReadModelEventsToEnvelopes,
-      fetch: fetchReadModel.bind(null, dynamoDB),
-      search: searchReadModel.bind(null, dynamoDB),
-      store: storeReadModel.bind(null, dynamoDB),
-      delete: deleteReadModel.bind(null, dynamoDB),
-      subscribe: subscribeToReadModel.bind(null, dynamoDB),
-      fetchSubscriptions: fetchSubscriptions.bind(null, dynamoDB),
-      deleteSubscription: deleteSubscription.bind(null, dynamoDB),
-      deleteAllSubscriptions: deleteAllSubscriptions.bind(null, dynamoDB),
-    },
-    // ProviderGraphQLLibrary
-    graphQL: {
-      rawToEnvelope: rawGraphQLRequestToEnvelope,
-      handleResult: requestSucceeded,
-    },
-    // ProviderAPIHandling
-    api: {
-      requestSucceeded,
-      requestFailed,
-    },
-    connections: {
-      storeData: storeConnectionData.bind(null, dynamoDB),
-      fetchData: fetchConnectionData.bind(null, dynamoDB),
-      deleteData: deleteConnectionData.bind(null, dynamoDB),
-      sendMessage: sendMessageToConnection,
-    },
-    // ScheduledCommandsLibrary
-    scheduled: {
-      rawToEnvelope: rawScheduledInputToEnvelope,
-    },
-    // ProviderInfrastructureGetter
-    infrastructure: () => {
-      const infrastructurePackageName = require('../package.json').name + '-infrastructure'
-      let infrastructure: HasInfrastructure | undefined
-      try {
-        infrastructure = loadInfrastructurePackage(infrastructurePackageName)
-      } catch (e) {
-        throw new Error(
-          `The AWS infrastructure package could not be loaded. The following error was thrown: ${e.message}. Please ensure that one of the following actions has been done:\n` +
-            `  - It has been specified in your "devDependencies" section of your "package.json" file. You can do so by running 'npm install --save-dev ${infrastructurePackageName}'\n` +
-            `  - Or it has been installed globally. You can do so by running 'npm install -g ${infrastructurePackageName}'`
-        )
-      }
-
-      return infrastructure.Infrastructure(rockets)
-    },
+  try {
+    require('aws-sdk')
+    const { Provider } = require('./setup.ts')
+    return Provider(rockets)
```

Comment:
For anybody wondering if this is the way to go, it is done following the [npm documentation](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#optionaldependencies)

Reply:
It's done to avoid issues when other providers try to load the `aws-sdk` dependency when the `AWS` provider is also set in the config file.

Reply:
Hey @alvaroloes! This was not the ideal technical solution, but it was the one that made sense from a business perspective. I'll send you some information about the issue so you can understand it better.